### ISSUE TITLE
LightGBM: four ignored parameters (`missing_type`, `zero_as_missing`, `sigmoid`, `reg_sqrt`)

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/tidypredict#288]
Four parameters that change LightGBM's own prediction are not consulted by the parser. Same file and same shape of fix, so grouping them.

- [ ] **`missing_type` is ignored, so NA routing is wrong for models trained without NA — the common case.** `R/model-lightgbm.R:314-330`, `:666-674`. tidypredict routes NA purely by `default_left`, but LightGBM's `Tree::NumericalDecision` consults `default_left` **only** when the node's `missing_type` is `NaN` (or `Zero`). When `missing_type == None` — which is what you get for *any feature with no NA in the training data* — LightGBM coerces NaN to `0.0` and compares it against the threshold normally.

  Repro: feature `neg ~ Uniform(-10,-5)`, regression, 6 rounds, no NA in training. All 21 splits on `neg` have `default_left = TRUE` and negative thresholds, so NaN→0 goes **right** in LightGBM and **left** in tidypredict:
  ```
  maxdiff = 8.250763
    lgb:         -10.4248  -11.05753  -10.91728
    tidypredict: -18.13758 -19.0575   -18.13758
  lgb pred for NA == lgb pred for 0 ?  diff = 0   # confirms the NaN->0 semantics
  ```
  Both the nested and v3 flat builders are affected. **The current tests miss this because they all train *with* NA** (`test-model-lightgbm.R:412,890,1590` use `use_missing = TRUE` on NA-containing data), which yields `missing_type = NaN` — the one case where `default_left` happens to be correct.

  `missing_type` is available per node in `model$dump_model()` but **not** in `lgb.model.dt.tree()`, so it has to be joined by `tree_index`/`split_index`. Then emit: `NaN` → current behaviour; `None` → `dplyr::coalesce(col, 0) <= t`, ignoring `default_left`; `Zero` → next item.

- [ ] **`zero_as_missing` is ignored.** With `zero_as_missing = TRUE` every node gets `missing_type = "Zero"`, and LightGBM routes **both NA and exact 0** to the `default_left` side; tidypredict compares zeros against the threshold. Regression with 40 zeros injected into `x2` plus a categorical feature is wrong **on the training data itself**: 1.30847, and the same on new zeros and on NA. Data-dependent — a simpler configuration gave 0. Fix: emit `(is.na(col) | col == 0) | col <= t` when `default_left`, else `!is.na(col) & col != 0 & col <= t`.

- [ ] **`reg_sqrt` is ignored.** `:435-453`, `lgb_identity_objectives` at `:3-11`. `objective = "regression"` with `reg_sqrt = TRUE` trains on `sqrt(|y|)`, and `RegressionL2loss::ConvertOutput` returns `sign(x) * x^2`; tidypredict treats it as identity. mtcars mpg, 5 rounds: **max abs diff 25.09**, larger than the response. Fix: in `apply_lgb_objective()`, when the objective is in `lgb_identity_objectives` and `isTRUE(params$reg_sqrt)`, return `expr(sign(!!f) * !!f^2)`; if the flag cannot be read reliably from a reloaded model, abort.

- [ ] **`sigmoid` is ignored.** `:370-372`, `:447-449`, `:393-394`. LightGBM's `binary`, `cross_entropy` (via `BinaryLogloss`) and `multiclassova` apply `1/(1+exp(-sigmoid * x))`; `lgb_sigmoid()` hardcodes 1. With `sigmoid = 2`: **max abs diff 0.1499** on a probability scale. Fix: pass `params$sigmoid %||% 1` and emit `1 / (1 + exp(-(s * f)))` on both the binary and `multiclassova` paths.

Two cleanups in the same area: `:436-440` is a dead `if` block with an empty body (the `rf` division happens in `assemble_lgb_formula()` at `:520-522`), and the `default_left == TRUE` branch for `decision_type == "=="` (`:336-341`, `:678-681`) is unreachable and would be wrong if reached — `CategoricalDecision` always sends NaN and negative values right, and `Tree::SplitCategorical` never sets the bit (confirmed empirically: 8 categorical splits, `default_left TRUE: 0`).

**Comment by EmilHvitfeldt** [comment]
All four parameters are now fixed and merged. Leaving this open for the two cleanups at the bottom, which both PRs deliberately skipped to stay reviewable.

**Done:**

- [x] `missing_type`, PR #347. Joined from the JSON dump by `tree_index`/`split_index`, since it is not in `lgb.model.dt.tree()`.
- [x] `zero_as_missing`, same PR. Both are `missing_type` in the end: `None` coerces a missing value to `0`, `Zero` treats an exact zero as missing.
- [x] `reg_sqrt`, PR #357.
- [x] `sigmoid`, same PR.

**Still open:** the dead `if` block at `:436-440`, and the unreachable `default_left == TRUE` branch for `decision_type == "=="` at `:336-341` and `:678-681`.

**A third fault came out of the `missing_type` sweep** that this issue does not mention: a categorical split carried a missing value left, which LightGBM never does. Fixed in #347 alongside the other two. 12 configurations by 3 seeds, all exact afterwards.

**Two corrections to the fixes as specified here,** both found by checking each objective against `predict()` rather than by reading the parameter list. Implemented as written, each would have been a regression:

- **`cross_entropy` accepts `sigmoid` but never applies it.** This issue lists it alongside `binary` as taking the scaling. It does not: its link is a plain logistic whatever the parameter says. Verified at sigmoid 1, 2 and 3, where `predict()` matches `logistic(raw)` exactly and `logistic(s * raw)` by 0.15 and 0.226. Scaling it would have broken a currently correct model.
- **`huber` accepts `reg_sqrt` but does not act on it.** The fix here applies the squaring to everything in `lgb_identity_objectives`. `huber` stays on the raw scale; squaring it is wrong by 90.7. The other five (`regression`, `regression_l2`, `regression_l1`, `fair`, `quantile`, `mape`) all honour it.

Both exceptions are now named constants in the source with the reason recorded next to them.
